main/core/src/mill/define/Cross.scala

```diff
@@ -24,6 +24,7 @@ object Cross {
           override def millOuterCtx = ctx0.copy(
             crossInstances = vs.map(v => new $tpe(..$argTupleValues))
           )
+          def millSourcePath: os.Path = millOuterCtx.millSourcePath
```

Comment:
Why is it needed?

Reply:
The overridden impl is

```scala
def millSourcePath: os.Path = millOuterCtx.millSourcePath / millOuterCtx.segment.pathSegments
```

As we don't want the cross matrix parameter end up in the path, we need to remove them.

Reply:
The issue with your initial solution is, that you removed the segment of the cross module but kept the segments of the cross instances. I changed it to keep the segment of the cross module, but remove the segments of the cross instances.

Unfortunately, this isn't a proper solution. Although it does the right thing, it breaks the ability to customize the `millSourcePath` in a cross module, as mix-in our override _after_ the users customization, we now loose any customized `millSourcePath`.

I think, I need to introduce another context value with always returns the desired segments, so we can use it in the default implementation of `Module.millSourcePath`.

Reply:
Change default `millSourcePath` to drop cross segments in 7c0edcc5f183886f886684064cb9712280d5835d. This might work...